src/OpenLoco/Windows/VehicleList.cpp

```diff
@@ -27,36 +51,850 @@ namespace OpenLoco::Ui::Windows::VehicleList
+                                   InterfaceSkin::ImageIds::vehicle_ships_frame_4,
+                                   InterfaceSkin::ImageIds::vehicle_ships_frame_5,
+                                   InterfaceSkin::ImageIds::vehicle_ships_frame_6,
+                                   InterfaceSkin::ImageIds::vehicle_ships_frame_7,
+                               } },
+        };
+
+        for (auto [tab, frames] : tabAnimations)
+        {
+            if (self->isDisabled(tab))
+                continue;
+
+            auto isActive = tab == self->current_tab + Widx::tab_trains;
+            auto imageId = isActive ? frames[self->frame_no / 2 % 8] : frames[0];
+
+            uint32_t image = Gfx::recolour(skin->img + imageId, companyColour);
+            Widget::draw_tab(self, dpi, image, tab);
+        }
+    }
+
+    // 0x004C28A5
+    static void disableUnavailableVehicleTypes(window* self)
+    {
+        // The original game looks at all companies here. We only look at the current company instead.
+        auto company = CompanyManager::get(self->number);
+
+        // Disable the tabs for the vehicles that are _not_ available for this company.
+        self->disabled_widgets = (company->available_vehicles ^ 0x3F) << Widx::tab_trains;
+    }
+
+    // 0x004C1AA2
+    static window* create(uint16_t companyId)
+    {
+        window* self = WindowManager::createWindow(
+            WindowType::vehicleList,
+            window_size,
+            WindowFlags::flag_11,
+            &_events);
+
+        self->widgets = _widgets;
+        self->enabled_widgets = _enabledWidgets;
+        self->number = companyId;
+        self->owner = companyId;
+        self->frame_no = 0;
+
+        auto skin = ObjectManager::get<InterfaceSkinObject>();
+        self->colours[1] = skin->colour_0A;
+
+        disableUnavailableVehicleTypes(self);
+
+        return self;
+    }
+
     // 0x004C19DC
     window* open(uint16_t companyId, uint8_t type)
     {
-        window* vehicle_list = WindowManager::bringToFront(WindowType::vehicleList, companyId);
-        if (vehicle_list == nullptr)
+        window* self = WindowManager::bringToFront(WindowType::vehicleList, companyId);
+        if (self != nullptr)
         {
-            registers regs;
-            regs.eax = (uint32_t)companyId;
-            regs.ebx = type;
-            call(0x004C1A05, regs);
-            vehicle_list = (window*)regs.esi;
+            self->callOnMouseUp(VehicleList::getTabFromType(type));
+            return self;
         }
+
+        initEvents();
+
+        // 0x004C1A05
+        self = create(companyId);
+        self->current_tab = type;
+        self->row_height = row_heights[type];
+        self->width = window_size.width;
+        self->height = window_size.height;
+        self->sort_mode = 0;
+        self->var_83C = 0;
+        self->row_hover = -1;
+
+        refreshVehicleList(self);
+
+        self->invalidate();
+
```

Comment:
why have the setting enabled, holdable, activated widgets been removed here?

Reply:
The enabled widgets are already set by `create`. There are no holdable widgets. The only activated widget (current tab) is set by prepareDraw.